gframe/deck_manager.cpp

```diff
@@ -436,15 +440,21 @@ bool DeckManager::SaveDeck(epro::path_stringview name, const cardlist_type& main
 		return false;
 	deckfile << "#created by " << BufferIO::EncodeUTF8(mainGame->ebNickName->getText()) << "\n#main\n";
 	for(auto card : mainlist)
-		deckfile << epro::to_string(card) << "\n";
+		deckfile << BuildCardLineToSave(card);
 	deckfile << "#extra\n";
 	for(auto card : extralist)
-		deckfile << epro::to_string(card) << "\n";
+		deckfile << BuildCardLineToSave(card);
 	deckfile << "!side\n";
 	for(auto card : sidelist)
-		deckfile << epro::to_string(card) << "\n";
+		deckfile << BuildCardLineToSave(card);
 	return true;
 }
+const std::string DeckManager::BuildCardLineToSave(uint32_t code) {
+	if (gGameConfig->addCardNamesToDeckList)
+		return epro::to_string(code) + "; " + gDataManager->GetNameString(code) + "\n";
+
+	return epro::to_string(code) + "\n";
+}
```

Comment:
This could be made a non class static function since it needs nothing from the DeckManager, maybe give it a better name as well, like `MakeYdkEntryString`

Reply:
Hmm, i think the same could be said about the DeckManager::ExportDeckBase64 function. If you really want this done then please specify where to put it because i don't quite get the project structure. I did not find a good place to put it.